File: src/main/java/com/giffardtechnologies/restdocs/domain/type/NamedType.java
package com.giffardtechnologies.restdocs.domain.type;

/**
 * An interface for types that have an identifying name.
 */
public interface NamedType {

	/**
	 * Returns the human-readable type name that servers as an identifier (unrelated to the data type).
	 *
	 * @return the human-readable type name
	 */
	String getTypeName();

	/**
	 * Returns the data type of this type, this will be {@link DataType#ENUM} or {@link DataType#OBJECT}
	 *
	 * @return the data type
	 */
	DataType getType();

}

File: src/main/java/com/giffardtechnologies/restdocs/domain/type/NamedType.kt
package com.giffardtechnologies.restdocs.domain.type

/**
 * An interface for types that have an identifying name.
 */
interface NamedType {
    /**
     * Returns the human-readable type name that servers as an identifier (unrelated to the data type).
     *
     * @return the human-readable type name
     */
    val typeName: String

    /**
     * Returns the [TypeSpec] for this type
     *
     * @return the data type
     */
    val type: TypeSpec.Nameable
}
